### PR TITLE
merge 'develop_v0.12' into 'release_v0.12'

### PR DESCRIPTION
**Pull request by omerspot**
* *New Resource*: `elastigroup_aws_suspend_processes`

* resource/spotinst_ocean_ecs_launch_spec: added `block_device_mappings`
* resource/spotinst_elastigroup_aws: added `batch` for `integration_ecs`
* resource/ocean_aws: resolved error with `auto_headroom_percentage` under `auto_scaler`.
* resource/spotinst_elastigroup_aws: resolved error with `target_scaling_policy` under `auto_scaler`.
* documentation: resource/spotinst_mrscaler_aws: fixed usage example and documentation for `retries`.

### CODE REVIEW

File: spotinst/elastigroup_aws_integrations/fields_spotinst_elastigroup_aws_integrations_ecs.go [hunk 1]
```diff
@@ -241,9 +256,45 @@ func expandAWSGroupEC2ContainerServiceIntegration(data interface{}) (*aws.EC2Con
...
+		batch, err := expandECSBatch(v)
+		if err != nil {
+			return nil, err
+		}
+		if batch != nil {
+			integration.SetBatch(batch)
+		} else {
+			integration.SetBatch(nil)
+		}
+	}
```
Comment (liranp) [inline spotinst/elastigroup_aws_integrations/fields_spotinst_elastigroup_aws_integrations_ecs.go:270]:
```go
batch, err := expandECSBatch(v)
if err != nil {
    return nil, err
}
if batch != nil {
    integration.SetBatch(batch)
} else {
    integration.SetBatch(nil)
}
```
You can simplify the code by removing the redundant `if-else` statement. You're calling `SetBatch` for both cases, so just use `integration.SetBatch(batch)`, for example:
```go
batch, err := expandECSBatch(v)
if err != nil {
    return nil, err
}
integration.SetBatch(batch)
```

File: spotinst/elastigroup_aws_integrations/fields_spotinst_elastigroup_aws_integrations_ecs.go [hunk 2]
```diff
@@ -241,9 +256,45 @@ func expandAWSGroupEC2ContainerServiceIntegration(data interface{}) (*aws.EC2Con
...
 	return integration, nil
 }
 
+func expandECSBatch(data interface{}) (*aws.Batch, error) {
+	if list := data.([]interface{}); len(list) > 0 {
+		batch := &aws.Batch{}
+		if list != nil && list[0] != nil {
+			m := list[0].(map[string]interface{})
+			var jobQueueNames []string = nil
+			if v, ok := m[string(JobQueueNames)].([]interface{}); ok && v != nil {
```
Comment (liranp) [inline spotinst/elastigroup_aws_integrations/fields_spotinst_elastigroup_aws_integrations_ecs.go:280]: Checking `if len(v) > 0` is [enough](https://play.golang.org/p/VAB7ehVMjRZ).

File: spotinst/elastigroup_aws_scaling_policies/fields_spotinst_elastigroup_aws_scaling_policies.go [hunk 3]
```diff
@@ -186,18 +186,19 @@ func baseScalingPolicySchema() *schema.Schema {
 				string(Statistic): {
 					Type:     schema.TypeString,
 					Optional: true,
-					Computed: true,
+					Default:  "average",
```
Comment (liranp) [inline spotinst/elastigroup_aws_scaling_policies/fields_spotinst_elastigroup_aws_scaling_policies.go:189]: Why did you add a default value?

File: spotinst/elastigroup_aws_scaling_policies/fields_spotinst_elastigroup_aws_scaling_policies.go [hunk 4]
```diff
@@ -186,18 +186,19 @@ func baseScalingPolicySchema() *schema.Schema {
...
 				},
 
 				string(Unit): {
 					Type:     schema.TypeString,
-					Required: true,
+					Optional: true,
+					Default:  "percent",
```
Comment (liranp) [inline spotinst/elastigroup_aws_scaling_policies/fields_spotinst_elastigroup_aws_scaling_policies.go:195]: Same here.

File: spotinst/elastigroup_aws_scaling_policies/fields_spotinst_elastigroup_aws_scaling_policies.go [hunk 5]
```diff
@@ -186,18 +186,19 @@ func baseScalingPolicySchema() *schema.Schema {
...
 				},
 
 				string(Cooldown): {
 					Type:     schema.TypeInt,
 					Optional: true,
-					Computed: true,
+					Default:  300,
```
Comment (liranp) [inline spotinst/elastigroup_aws_scaling_policies/fields_spotinst_elastigroup_aws_scaling_policies.go:201]: Same here.

File: spotinst/elastigroup_aws_strategy/fields_spotinst_elastigroup_aws_strategy.go [hunk 6]
```diff
@@ -18,35 +18,40 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
 			egWrapper := resourceObject.(*commons.ElastigroupWrapper)
 			elastigroup := egWrapper.GetElastigroup()
+			//Risk is configured as int in the API but float on Go-SDK (currently not aligning because of breaking changes effects)
+			//There for value is of type float and cast as necessary
 			var value *float64 = nil
 			if elastigroup.Strategy != nil && elastigroup.Strategy.Risk != nil {
 				value = elastigroup.Strategy.Risk
 			}
-			if err := resourceData.Set(string(SpotPercentage), spotinst.Float64Value(value)); err != nil {
+			//Casting from float to int
```
Comment (liranp) [inline spotinst/elastigroup_aws_strategy/fields_spotinst_elastigroup_aws_strategy.go:34]: I believe we should omit comments of this kind (it's clear from the code itself).

File: spotinst/elastigroup_aws_strategy/fields_spotinst_elastigroup_aws_strategy.go [hunk 7]
```diff
@@ -18,35 +18,40 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
 			}
 			return nil
 		},
 		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
 			egWrapper := resourceObject.(*commons.ElastigroupWrapper)
 			elastigroup := egWrapper.GetElastigroup()
-			if v, ok := resourceData.Get(string(SpotPercentage)).(float64); ok && v >= 0 {
-				elastigroup.Strategy.SetRisk(spotinst.Float64(v))
+			if v, ok := resourceData.Get(string(SpotPercentage)).(int); ok && v >= 0 {
+				//Casting from int to float
```
Comment (liranp) [inline spotinst/elastigroup_aws_strategy/fields_spotinst_elastigroup_aws_strategy.go:44]: Same here.

File: spotinst/elastigroup_aws_strategy/fields_spotinst_elastigroup_aws_strategy.go [hunk 8]
```diff
@@ -18,35 +18,40 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
```
Comment (liranp) [inline spotinst/elastigroup_aws_strategy/fields_spotinst_elastigroup_aws_strategy.go:53]: Same here.

File: spotinst/elastigroup_aws_suspend_processes/consts.go [hunk 9]
```diff
@@ -0,0 +1,9 @@
+package elastigroup_aws_suspend_processes
+
+import "github.com/terraform-providers/terraform-provider-spotinst/spotinst/commons"
+
+const (
+	GroupID     commons.FieldName = "group_id"
+	Suspensions commons.FieldName = "suspension"
```
Comment (liranp) [inline spotinst/elastigroup_aws_suspend_processes/consts.go:7]: The name of the const (`Suspensions`) is plural while the value (`suspension `) is singular.

```suggestion
	Suspension commons.FieldName = "suspension"
```

File: spotinst/elastigroup_aws_suspend_processes/fields_suspend_processes.go [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+		&schema.Schema{
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			spWrapper := resourceObject.(*commons.SuspendProcessesWrapper)
+
+			var value *string = nil
+			value = spotinst.String(resourceData.Get(string(GroupID)).(string))
```
Comment (liranp) [inline spotinst/elastigroup_aws_suspend_processes/fields_suspend_processes.go:29]:
```suggestion
			value := spotinst.String(resourceData.Get(string(GroupID)).(string))
```

File: spotinst/elastigroup_aws_suspend_processes/fields_suspend_processes.go [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+		GroupID,
+		&schema.Schema{
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			spWrapper := resourceObject.(*commons.SuspendProcessesWrapper)
+
+			var value *string = nil
```
Comment (liranp) [inline spotinst/elastigroup_aws_suspend_processes/fields_suspend_processes.go:28]: Make the change in the next line and remove this one.

```suggestion

```

File: spotinst/ocean_aws_strategy/fields_spotinst_ocean_aws_strategy.go [hunk 12]
```diff
@@ -116,38 +116,43 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
 			//Force setting -1 as default value if it's not exists in initial creation,
 			// to allow initialization of the field to 0
+			//SpotPercentage is configured as int in the API but float on Go-SDK (currently not aligning because of breaking changes effects)
+			//There for value is of type float and cast as necessary
 			value := spotinst.Float64(-1)
 			if cluster.Strategy != nil && cluster.Strategy.SpotPercentage != nil {
 				value = cluster.Strategy.SpotPercentage
 			}
-			if err := resourceData.Set(string(SpotPercentage), spotinst.Float64Value(value)); err != nil {
+			//Casting from float to int
```
Comment (liranp) [inline spotinst/ocean_aws_strategy/fields_spotinst_ocean_aws_strategy.go:135]: [Same comment](https://github.com/spotinst/terraform-provider-spotinst/pull/246/files#r499260839).

File: spotinst/ocean_aws_strategy/fields_spotinst_ocean_aws_strategy.go [hunk 13]
```diff
@@ -116,38 +116,43 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
 			}
 			return nil
 		},
 		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
 			clusterWrapper := resourceObject.(*commons.AWSClusterWrapper)
 			cluster := clusterWrapper.GetCluster()
-			if v := resourceData.Get(string(SpotPercentage)).(float64); v > -1 {
-				cluster.Strategy.SetSpotPercentage(spotinst.Float64(v))
+			if v := resourceData.Get(string(SpotPercentage)).(int); v > -1 {
+				//Casting from int to float
```
Comment (liranp) [inline spotinst/ocean_aws_strategy/fields_spotinst_ocean_aws_strategy.go:154]: [Same comment](https://github.com/spotinst/terraform-provider-spotinst/pull/246/files#r499260839).

File: spotinst/ocean_aws_strategy/fields_spotinst_ocean_aws_strategy.go [hunk 14]
```diff
@@ -116,38 +116,43 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
```
Comment (liranp) [inline spotinst/ocean_aws_strategy/fields_spotinst_ocean_aws_strategy.go:145]: [Same comment](https://github.com/spotinst/terraform-provider-spotinst/pull/246/files#r499260839).

File: spotinst/ocean_ecs_launch_spec/fields_spotinst_ocean_ecs_launch_spec.go [hunk 15]
```diff
@@ -488,6 +488,155 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
+			return nil
+		},
+
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			LaunchSpecWrapper := resourceObject.(*commons.ECSLaunchSpecWrapper)
+			launchSpec := LaunchSpecWrapper.GetLaunchSpec()
+			var value []*aws.ECSBlockDeviceMapping = nil
+
+			if v, ok := resourceData.GetOk(string(BlockDeviceMappings)); ok {
+				if blockdevicemappings, err := expandBlockDeviceMappings(v); err != nil {
```
Comment (liranp) [inline spotinst/ocean_ecs_launch_spec/fields_spotinst_ocean_ecs_launch_spec.go:628]:
```suggestion
				if blockDeviceMappings, err := expandBlockDeviceMappings(v); err != nil {
```

File: spotinst/ocean_ecs_launch_spec/fields_spotinst_ocean_ecs_launch_spec.go [hunk 16]
```diff
@@ -488,6 +488,155 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
+					return err
+				} else {
+					value = blockdevicemappings
```
Comment (liranp) [inline spotinst/ocean_ecs_launch_spec/fields_spotinst_ocean_ecs_launch_spec.go:631]:
```suggestion
					value = blockDeviceMappings
```

File: spotinst/ocean_ecs_launch_spec/fields_spotinst_ocean_ecs_launch_spec.go [hunk 17]
```diff
@@ -643,3 +792,157 @@ func flattenTags(tags []*aws.Tag) []interface{} {
...
 }
+
+func expandBlockDeviceMappings(data interface{}) ([]*aws.ECSBlockDeviceMapping, error) {
+
+	list := data.([]interface{})
+	bdms := make([]*aws.ECSBlockDeviceMapping, 0, len(list))
+
+	for _, v := range list {
+		attr, ok := v.(map[string]interface{})
+		bdm := &aws.ECSBlockDeviceMapping{}
```
Comment (liranp) [inline spotinst/ocean_ecs_launch_spec/fields_spotinst_ocean_ecs_launch_spec.go:803]: Move to line 808. There's no reason to declare a variable before checking the value of `ok`.

File: spotinst/provider.go [hunk 18]
```diff
@@ -64,6 +64,9 @@ func Provider() terraform.ResourceProvider {
 
 			// HealthCheck
 			string(commons.HealthCheckResourceName): resourceSpotinstHealthCheck(),
+
+			//suspendProcesses
```
Comment (liranp) [inline spotinst/provider.go:68]:
```suggestion
			// SuspendProcesses
```

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go [hunk 19]
```diff
@@ -0,0 +1,283 @@
+package spotinst
+
+import (
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:3]: Please execute `goimports -w ./spotinst` to format all imports.

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go [hunk 20]
```diff
@@ -0,0 +1,283 @@
...
+
+func setupSuspendProcesses() {
+	fieldsMap := make(map[commons.FieldName]*commons.GenericField)
+
+	elastigroup_aws_suspend_processes.Setup(fieldsMap)
+
+	commons.SuspendProcessesResource = commons.NewSuspendProcessesResource(fieldsMap)
+}
+
+////-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:43]:
```suggestion
//-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
```

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go [hunk 21]
```diff
@@ -0,0 +1,283 @@
...
+////          Read
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:44]:
```suggestion
//          Read
```

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go [hunk 22]
```diff
@@ -0,0 +1,283 @@
...
+////-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:45]:
```suggestion
//-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
```

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go [hunk 23]
```diff
@@ -0,0 +1,283 @@
...
+
+	if err := commons.SuspendProcessesResource.OnRead(spResponse.SuspendProcesses, resourceData, meta); err != nil {
+		return err
+	}
+
+	log.Printf("===> Elastigroup read successfully: %s <===", resourceData.Id())
+	return nil
+}
+
+////-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:91]: [Same comment](https://github.com/spotinst/terraform-provider-spotinst/pull/246/files#r499261947).

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go [hunk 24]
```diff
@@ -0,0 +1,283 @@
...
+
+		if _, err := meta.(*Client).elastigroup.CloudProviderAWS().DeleteSuspensions(context.Background(), deleteReqBody); err != nil {
+			return fmt.Errorf("[ERROR] onDelete() -> Failed to delete suspendProcesses for Elastigroup: %s", err)
+		}
+	}
+
+	return nil
+}
+
+func Find(slice []string, val string) (int, bool) {
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:276]: Why do you export that function? Also, `Find` is too generic, please use a more descriptive name, like `findString`.

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes_test.go [hunk 25]
```diff
@@ -0,0 +1,291 @@
+package spotinst
+
+import (
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes_test.go:3]: Format the imports. Did you execute `make fmt`?

File: version/version.go [hunk 26]
```diff
@@ -10,7 +10,7 @@ var (
 	// Version represents the main version number.
 	//
 	// Read-only.
-	Version = "1.25.0"
+	Version = "1.27.0"
```
Comment (liranp) [inline version/version.go:13]: What happened to version `1.26.0`?

File: website/docs/r/elastigroup_aws_suspend_processes.html.markdown [hunk 27]
```diff
@@ -0,0 +1,40 @@
...
+# spotinst\_elastigroup\_aws\_susoend_processes
+
+Suspend AWS Elastigroup processes. This resource provide the capavility of
+suspending elastigroup processes using Terraform.
+
+For supported processes please visit: [Suspend Processes API reference](https://help.spot.io/spotinst-api/elastigroup/amazon-web-services/suspend-processes/)
+## Example Usage
+
+```hcl
+# Create an Elastigroup
```
Comment (liranp) [inline website/docs/r/elastigroup_aws_suspend_processes.html.markdown:18]: Please update the comment.

File: spotinst/commons/common_elastigroup_aws_suspend_processes.go [hunk 28]
```diff
@@ -0,0 +1,129 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/spotinst/spotinst-sdk-go/service/elastigroup/providers/aws"
+	"log"
+)
+
+//-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+//            Variables
+//-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+const (
+	SuspendProcessesResourceName ResourceName = "spotinst_elastigroup_aws_suspend_processes"
```
Comment (liranp) [inline spotinst/commons/common_elastigroup_aws_suspend_processes.go:14]: I believe `spotinst_elastigroup_aws_suspension` is a more suitable name. Resource names should be nouns since a resource block represents an object rather than an action. For me, `[...]_suspend_processes` sounds like a verb which represents an action. @nadavnaor WDYT?

File: spotinst/elastigroup_aws_scaling_policies/fields_spotinst_elastigroup_aws_scaling_policies.go (hunk 3 above)
Comment (omerspot) [reply spotinst/elastigroup_aws_scaling_policies/fields_spotinst_elastigroup_aws_scaling_policies.go:189]: To reflect changes made in the optimizer, that caused the "plan" scheme to show that changes were made even though there aren't any (so "applying" and "planning" right after will show changes although nothing changed)

File: spotinst/elastigroup_aws_scaling_policies/fields_spotinst_elastigroup_aws_scaling_policies.go (hunk 4 above)
Comment (omerspot) [reply spotinst/elastigroup_aws_scaling_policies/fields_spotinst_elastigroup_aws_scaling_policies.go:195]: Same [here](https://github.com/spotinst/terraform-provider-spotinst/pull/246#discussion_r499392446)

File: spotinst/elastigroup_aws_scaling_policies/fields_spotinst_elastigroup_aws_scaling_policies.go (hunk 5 above)
Comment (omerspot) [reply spotinst/elastigroup_aws_scaling_policies/fields_spotinst_elastigroup_aws_scaling_policies.go:201]: Same [here](https://github.com/spotinst/terraform-provider-spotinst/pull/246#discussion_r499392446)

File: spotinst/commons/common_elastigroup_aws_suspend_processes.go (hunk 28 above)
Comment (omerspot) [reply spotinst/commons/common_elastigroup_aws_suspend_processes.go:14]: @liranp, We will make the change. We have discussed with @nadavnaor  and he agreed as well.

File: spotinst/elastigroup_aws_integrations/fields_spotinst_elastigroup_aws_integrations_ecs.go (hunk 1 above)
Comment (Lironrad) [reply spotinst/elastigroup_aws_integrations/fields_spotinst_elastigroup_aws_integrations_ecs.go:270]: We need to explicitly send nil object to reset the object to a null object. If we won’t do so, terraform does not update this field and consider it as unchanged field.

For example -
When we try to update from:

```
integration_ecs {
  cluster_name         = "ecs-cluster"
  autoscale_is_enabled = false
  autoscale_cooldown   = 300
  autoscale_scale_down_non_service_tasks = false

  autoscale_headroom {
    cpu_per_unit    = 1024
    memory_per_unit = 512
    num_of_units    = 2
  }

  autoscale_down {
    evaluation_periods        = 300
    max_scale_down_percentage = 70
  }

  autoscale_attributes {
    key   = "test.ecs.key"
    value = "test.ecs.value"
  }

  batch{
    job_queue_names = [
      "job-1",
      "job-2"
    ]
  }
}

```
to:
```
integration_ecs {
  cluster_name         = "ecs-cluster"
  autoscale_is_enabled = false
  autoscale_cooldown   = 300
  autoscale_scale_down_non_service_tasks = false

  autoscale_headroom {
    cpu_per_unit    = 1024
    memory_per_unit = 512
    num_of_units    = 2
  }

  autoscale_down {
    evaluation_periods        = 300
    max_scale_down_percentage = 70
  }

  autoscale_attributes {
    key   = "test.ecs.key"
    value = "test.ecs.value"
  }
}
```
We will get the same configuration as the first one.
Same problem occur also in the trying set auto_headroom_percentege to 0 [ticket](https://spotinst.atlassian.net/browse/CA-7775).

@liranp 
Am I missing something?  Do you think this is a behavior worth covering?

Comment (liranp) [reply spotinst/elastigroup_aws_integrations/fields_spotinst_elastigroup_aws_integrations_ecs.go:270]: You're calling `SetBatch` in both cases, so the `if-else` statement is redundant here. `SetBatch(batch)` will cover both cases where `expandECSBatch` returns either an object or a `nil`.

```go
batch, err := expandECSBatch(v)
// [...]
if batch != nil {
    integration.SetBatch(batch)
} else {
    integration.SetBatch(nil)
}
```

is equivalent to:

```go
batch, err := expandECSBatch(v)
// [...]
integration.SetBatch(batch)
```

Comment (Lironrad) [reply spotinst/elastigroup_aws_integrations/fields_spotinst_elastigroup_aws_integrations_ecs.go:270]: OK. Thanks for the clarification!

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go [hunk 29]
```diff
@@ -0,0 +1,283 @@
...
+
+func deleteSuspendProcesses(resourceData *schema.ResourceData, meta interface{}) error {
+
+	listInput := &aws.ListSuspensionsInput{}
+	gID := resourceData.Id()
+	listInput.GroupID = &gID
+
+	curr, err := meta.(*Client).elastigroup.CloudProviderAWS().ListSuspensions(context.Background(), listInput)
+	if err != nil {
+		return fmt.Errorf("[ERROR] failed to update suspend processes")
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:165]: Please expose the upstream error.

```suggestion
		return fmt.Errorf("[ERROR] failed to update suspend processes: %v", err)
```

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go [hunk 30]
```diff
@@ -0,0 +1,283 @@
...
+
+	var processesToDelete []string = nil
+
+	req := &aws.ListSuspensionsInput{}
+	gID := resourceData.Id()
+	req.GroupID = &gID
+
+	curr, err := meta.(*Client).elastigroup.CloudProviderAWS().ListSuspensions(context.Background(), req)
+	if err != nil {
+		return fmt.Errorf("[ERROR] failed to update suspend processes")
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:226]: Same here.

```suggestion
		return fmt.Errorf("[ERROR] failed to update suspend processes: %v", err)
```

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go [hunk 31]
```diff
@@ -0,0 +1,283 @@
...
+	}
+
+	if processesToCreate != nil {
+		createReqBody := &aws.CreateSuspensionsInput{Suspensions: suspendProcesses.Suspensions}
+		groupIDInput := resourceData.Id()
+		createReqBody.GroupID = &groupIDInput
+
+		_, err = meta.(*Client).elastigroup.CloudProviderAWS().CreateSuspensions(context.Background(), createReqBody)
+		if err != nil {
+			return fmt.Errorf("[ERROR] failed to update suspend processes")
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go:258]: Same here.

```suggestion
			return fmt.Errorf("[ERROR] failed to update suspend processes: %v, err")
```

File: spotinst/resource_spotinst_elastigroup_aws_suspend_processes_test.go [hunk 32]
```diff
@@ -0,0 +1,291 @@
...
+func testSuspendProcessesDestroy(s *terraform.State) error {
+	client := testAccProviderAWS.Meta().(*Client)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != string(commons.SuspendProcessesResourceName) {
+			continue
+		}
+		input := &aws.DeleteSuspensionsInput{GroupID: spotinst.String(rs.Primary.ID)}
+		resp, err := client.elastigroup.CloudProviderAWS().DeleteSuspensions(context.Background(), input)
+		if err == nil && resp != nil {
+			return fmt.Errorf("Suspend Processes still exists")
```
Comment (liranp) [inline spotinst/resource_spotinst_elastigroup_aws_suspend_processes_test.go:28]:
```suggestion
			return fmt.Errorf("suspend processes still exist")
```
